### PR TITLE
Rain mode with frequency

### PR DESCRIPTION
**Pull request by e2002e**
Hello, I recently found a way to have the 'rain' technique take into account the position of the input character set.
I tested it with success beating -inc mode with lower set (with the hashlist I used, at least).
Here are the actual records of the run:

Digits: 30 seconds -max-len=22
Inc 384g
Rain 123g
Inc | Rain 22g

Lowernum: 30 seconds -max-len=22
Inc 79
Rain 4
Inc | Rain 4

Lower: 60 seconds -max-len=15
Inc 35
Rain 57
Inc | Rain 57
All the tests are done with corresponding mask on the GPU.
I once more leave it up to you.

### CODE REVIEW

**Comment by magnumripper** [comment]
You really should learn how to rebase and squash commits. Make that your highest prio for a short while, it's not rocket science.

File: src/options.c [hunk 1]
```diff
@@ -292,10 +296,22 @@ PRINCE_USAGE \
...
+"--stdout[=LENGTH]          just output candidate passwords [cut at LENGTH]\n" \
+"--restore[=NAME]           restore an interrupted session [called NAME]\n" \
+"--session=NAME             give a new session the NAME\n" \
+"--status[=NAME]            print status of a session [called NAME]\n" \
+"--make-charset=FILE        make a charset file. It will be overwritten\n" \
+"--reject-printable         reject printable binaries\n" \
+"--show[=left]              show cracked passwords [if =left, then uncracked]\n" \
+"--show=formats             show information about hashes in a file (JSON)\n" \
+"--show=invalid             show lines that are not valid for selected format(s)\n" \
+"--test[=TIME]              run tests and benchmarks for TIME seconds each\n" \
```
Comment (magnumripper) [inline src/options.c:314]: Very obvious merge conflict here (but we wouldn't merge it anyway)

File: src/options.h [hunk 2]
```diff
@@ -186,7 +186,13 @@
 /* Configure terminal for reading keystrokes even if we're not the foreground process */
 #define FLG_FORCE_TTY			0x2000000000000000ULL
 /* Turn off logging */
-#define FLG_NOLOG			0x4000000000000000ULL
+#define FLG_NOLOG				0x4000000000000000ULL
+
```
Comment (magnumripper) [inline src/options.h:190]: You are changing unrelated code here

**Comment by e2002e** [comment]
I though I had rebased, but yes I see all these commits.... Doing this now
You tell me if that's rebased or not. I did a fetch with upstream to your repo, a pull, nothing was to rebase then. Even squashing commits it tells me it's uptodate.
I'm quite lost. Please test it to see if you have any interest before I start putting my head into it.

**Review by magnumripper — CHANGES_REQUESTED** [review]
I can't see us merging this at any time in any possible future, but I reviewed it anyway in case it would be educating for me or you.

Stop wasting your time on this.

**Comment by e2002e** [comment]
> Stop wasting your time on this.
I actually left that alone for a long time, took me a bit but that was no time...
No drama, I think this is good but king david from israel said "every man is right in his own eyes".
I am leaving this project untill you give me a definitive answer.
